dbschema/project.esdl

```diff
@@ -259,4 +358,44 @@ module Project {
           ))
       select reportPeriodRanges
   )
+
+  # TODO - Toying with the idea of abstracting some of this logic in some capacity...
```

Comment:
@CarsonF - After the trigger code settles down I think there's some opportunity for some common methods like I started here.

Reply:
I don't think functions can mutate data right now. I don't really understand the limitation.